core/common/src/files/Paths.kt

```diff
@@ -5,29 +5,118 @@
 
 package kotlinx.io.files
 
-import kotlinx.io.Sink
-import kotlinx.io.Source
+import kotlinx.io.*
+import kotlin.jvm.JvmName
 
 /**
  * A wrapper around a string representing a file path allowing to read from and write to a
  * corresponding file using [Sink] and [Source].
  *
  * **This API is unstable and subject to change.**
  */
-public expect class Path
+public expect class Path {
+    /**
+     * Returns a path representing a parent directory for this path,
+     * or `null` if there is no parent directory for this path.
+     *
+     * How the parent path is resolved is platform-specific.
+     */
+    public val parent: Path?
+
+    /**
+     * Returns the name of the file or directory represented by this path.
+     *
+     * How the name is resolved is platform-specific.
+     *
+     * In general, one may expect that for a path like `Path("home", "Downloads", "file.txt")`
+     * the name is `file.txt`.
+     */
+    public val name: String
+
+    /**
+     * Returns `true` if this path is absolute, `false` otherwise.
+     *
+     * How an absolute path is resolved is platform-specific.
+     */
+    public val isAbsolute: Boolean
+
+    /**
+     * Returns a string representation of this path.
+     *
+     * Note that the returned value will represent the same path as the value
+     * passed to [Path], but it may not be identical to it.
+     */
+    override fun toString(): String
+
+    override fun hashCode(): Int
```

Comment:
Also needs suppress or documentation.
Taking into account we already have `equals` specified, `hashCode` has more or less a single way to be specified as well